src/meta/ActiveHostsMan.h

```diff
@@ -34,95 +37,53 @@ struct HostInfo {
     int64_t lastHBTimeInSec_ = 0;
 };
 
-class ActiveHostsMan;
-
-class ActiveHostsManHolder final {
-public:
-    ActiveHostsManHolder() = delete;
-    ~ActiveHostsManHolder() = delete;
-
-    static ActiveHostsMan* hostsMan() {
-       static auto hostsMan
-           = std::make_unique<ActiveHostsMan>(FLAGS_expired_hosts_check_interval_sec,
-                                              FLAGS_expired_threshold_sec);
-       return hostsMan.get();
-    }
-};
+using UpdateCallback = std::function<void(IPv4 ip, Port port)>;
+using CleanCallback = std::function<void(const std::vector<HostAddr>&)>;
 
 class ActiveHostsMan final {
     FRIEND_TEST(ActiveHostsManTest, NormalTest);
 
 public:
-    ActiveHostsMan(int32_t intervalSeconds, int32_t expiredSeconds)
-        : intervalSeconds_(intervalSeconds)
-        , expirationInSeconds_(expiredSeconds) {
-        CHECK_GT(intervalSeconds, 0)
-            << "intervalSeconds " << intervalSeconds << " should > 0!";
-        CHECK_GE(expiredSeconds, intervalSeconds)
-            << "expiredSeconds " << expiredSeconds
-            << " should >= intervalSeconds " << intervalSeconds;
-        CHECK(checkThread_.start());
-        checkThread_.addTimerTask(intervalSeconds * 1000,
-                                  intervalSeconds * 1000,
-                                  &ActiveHostsMan::cleanExpiredHosts,
-                                  this);
+    static ActiveHostsMan* instance() {
+        static ActiveHostsMan activeHostsMan(FLAGS_expired_hosts_check_interval_sec,
```

Comment:
We'd better use a unique_ptr to hold the instance,  so it could be released when the process shut down.